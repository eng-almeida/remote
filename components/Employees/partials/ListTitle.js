import Link from 'next/link';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Button from '@/components/Elements/Button';
import { UserIcon } from '@/icons/index';

const Container = styled.div`
  align-items: center;
  margin: ${({ theme }) => `${theme.space[12]} auto ${theme.space[13]} auto`};
  display: flex;
  justify-content: space-between;
`;

const Title = styled.h1`
  color: ${({ theme }) => theme.colors.grays[300]};
  display: inline-block;
  font-weight: ${({ theme }) => theme.fontWeights.medium};
  font-size: 26px;
  line-height: 31px;
  margin-right: ${({ theme }) => theme.space[2]};
`;

const EmployeesCount = styled.span`
  color: ${({ theme }) => theme.colors.grays[100]};
  display: inline-block;
  font-weight: ${({ theme }) => theme.fontWeights.medium};
  font-size: 13px;
  line-height: 16px;
`;

const ListTitle = ({ employeesCount }) => (
  <Container>
    <div>
      <Title>People</Title>
      <EmployeesCount>{employeesCount} employees</EmployeesCount>
    </div>
    <Link href="/employee" passHref>
      <Button as="a">
        <UserIcon />
        Add employee
      </Button>
    </Link>
  </Container>
);

ListTitle.propTypes = {
  employeesCount: PropTypes.number,
};

export default ListTitle;
