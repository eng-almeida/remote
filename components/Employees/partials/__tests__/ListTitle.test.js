import ListTitle from '../ListTitle';

describe('ListTitle component', () => {
	it('should match snapshot', () => {
		const { container } = renderWithProviders(
			<ListTitle employeesCount={3}	/>
		);
		expect(container).toMatchSnapshot();
	});
})