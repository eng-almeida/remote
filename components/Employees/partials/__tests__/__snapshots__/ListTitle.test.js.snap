// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTitle component should match snapshot 1`] = `
.c3 {
  display: inline-block;
  text-align: center;
  background: #624DE3;
  border: 2px solid #624DE3;
  border-radius: 50px;
  box-shadow: 0px 6px 12px rgba(98,77,227,0.3);
  padding: 12px 26px;
  color: #ffffff;
  font-size: 16px;
  line-height: 19px;
  font-family: inherit;
  min-width: 142px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 48px auto 56px auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  color: #00234B;
  display: inline-block;
  font-weight: 500;
  font-size: 26px;
  line-height: 31px;
  margin-right: 8px;
}

.c2 {
  color: #778CA3;
  display: inline-block;
  font-weight: 500;
  font-size: 13px;
  line-height: 16px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <h1
        class="c1"
      >
        People
      </h1>
      <span
        class="c2"
      >
        3
         employees
      </span>
    </div>
    <a
      class="c3"
      href="/employee"
    >
      Add employee
    </a>
  </div>
</div>
`;
